node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/DiskReplacer.java

```diff
@@ -33,26 +42,41 @@ public class DiskReplacer extends NodeRepositoryMaintainer {
     @Override
     protected double maintain() {
         NodeList nodes = nodeRepository().nodes().list().rebuilding(true);
+        int attempts = 0;
         int failures = 0;
-        for (var host : nodes) {
-            Optional<NodeMutex> optionalMutex = nodeRepository().nodes().lockAndGet(host, Duration.ofSeconds(10));
-            if (optionalMutex.isEmpty()) continue;
-            try (NodeMutex mutex = optionalMutex.get()) {
-                // Re-check flag while holding lock
-                host = mutex.node();
-                if (!host.status().wantToRebuild()) {
-                    continue;
+        try (var locked = nodeRepository().nodes().lockAndGetAll(nodes.asList(), Optional.of(Duration.ofSeconds(10)))) {
```

Comment:
Cap the list to e.g. 10?

Reply:
But I guess something else stops these instances ... and attempting the first 10 over and over due to some failure would keep all later hosts waiting indefinitely.